Makefile

```diff
@@ -177,10 +177,21 @@ test-odo-login-e2e:
 test-odo-config:
 	ginkgo -v -nodes=$(TEST_EXEC_NODES) -focus="odo config test" slowSpecThreshold=$(SLOW_SPEC_THRESHOLD) -randomizeAllSpecs  tests/integration/ -timeout $(TIMEOUT)
 
-# Run all integration tests
-.PHONY: test-integration
-test-integration:
-	go test -v github.com/openshift/odo/tests/integration -ginkgo.slowSpecThreshold=$(SLOW_SPEC_THRESHOLD) -ginkgo.v -timeout $(TIMEOUT)
+# Runs all integration test irrespective of service catalog status in the cluster.
+# Service and link command tests are not the part of this test run
+.PHONY: test-integration-all
+test-integration-all:
+	ginkgo -v -nodes=$(TEST_EXEC_NODES) \
+	-focus="odoCmdApp|odoCmpSubE2e|odoCmpE2e|odo config test|odo storage command|odoWatchE2e|odo generic|odoJavaE2e|odojsonoutput|odo push command tests|odoURLIntegration|odoSourceE2e" \
```

Comment:
If a new integration test is added we will have to remember to include it in this long list. I'm afraid that this will lead to ugly errors where we forget to include it here.

Also, how can we be sure that all tests are really executed?

Reply:
> If a new integration test is added we will have to remember to include it in this long list. I'm afraid that this will lead to ugly errors where we forget to include it here.

Yup there is a high chance of forgetting new test name addition in focus. However i have a plan and that is almost implemented in my very next pr #1878 which i am going to open for review in few days.

Plan to overcome this disadvantage:

The top level test description in the test file should be in the form of ```odo <test command> command tests``` (For example - ```odo storage command tests```), so that the regular expression ```odo\s.*\scommand\stests``` passed through the -focus flag will take care of it. 
 

> Also, how can we be sure that all tests are really executed?

This is a ginkgo feature of ```-focus``` flag. It match the re expression passed in -focus flag and execute those. See https://github.com/openshift/odo/pull/1912#issuecomment-515866554